File: AtCoder/Greens/python/ARC026-B-Kanzensuu.py
N = int(input())


def divisors():
    res = []
    for i in range(1, int(N ** 0.5) + 1):
        if N % i == 0:
            res.append(i)
            if N // i != i:
                res.append(N // i)
    return res


divs = divisors()
sm = sum(divs) - N
if sm == N:
    print("Perfect")
elif sm < N:
    print("Deficient")
else:
    print("Abundant")

File: AtCoder/Greens/cpp/ARC026-B-Kanzensuu.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

ll N;

vector<ll> divisors() {
  vector<ll> res;
  for (ll i = 1; i * i <= N; ++i) {
    if (N % i == 0) {
      res.push_back(i);
      if (N / i != i) { res.push_back(N / i); }
    }
  }
  return res;
}

int main() {
  cin >> N;
  vector<ll> divs = divisors();
  ll sum = 0;
  for (auto v : divs) {
    if (v != N) sum += v;
  }
  if (sum == N)
    puts("Perfect");
  else if (sum < N)
    puts("Deficient");
  else
    puts("Abundant");
}